gpu/octree/src/cuda/radius_search.cu

```diff
@@ -237,7 +241,9 @@ namespace pcl
 
                     int length = end - beg;
 
-                    int *out = batch.output + active_query_index * batch.max_results + active_found_count;                    
+                    const std::ptrdiff_t result_offset = active_query_index * batch.max_results + active_found_count;
+                    int *out = batch.output + result_offset;
+                    float *sqr_dist = batch.sqr_distances + result_offset;
                     const int length_left = batch.max_results - active_found_count;
 
                     const int test = __any_sync(0xFFFFFFFF, active_lane == laneId && (leaf & KernelPolicy::CHECK_FLAG));
```

Comment:
Can we find a better name than 'test' here?

Reply:
`thread_active`?

Reply:
Im on mobile, so hard to fully grasp the ` (leaf & KernelPolicy::CHECK_FLAG` check, does this check if we are at a leaf?

So: thread_active_leaf?

Reply:
adjective before noun please. `active_thread`.

Reply:
> Im on mobile, so hard to fully grasp the (leaf & KernelPolicy::CHECK_FLAG check, does this check if we are at a leaf?

Actually from what I gather `(leaf & KernelPolicy::CHECK_FLAG)` is checking if the leaf should be checked (i.e. whether the leaf falls entirely inside the radius, and thus its contents can be copied directly, or if its only partially in the radius, implying that its contents needs to be checked). The other part is checking whether there is an active lane. All in all it's kind of checking whether a leaf must be processed.